product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -10,6 +10,125 @@
 class ProductProduct(models.Model):
     _inherit = 'product.product'
 
+    @api.multi
+    def name_get(self):
+
+        def _name_get(d):
+            name = d.get('name', '')
+            code = self._context.get('display_default_code', True) and d.get(
+                'default_code', False) or False
+            if code:
+                name = '[%s] %s' % (code, name)
+            return d['id'], name
+
+        partner_id = self._context.get('partner_id')
+        if partner_id:
+            partner_ids = [partner_id, self.env['res.partner'].browse(
+                partner_id).commercial_partner_id.id]
+        else:
+            partner_ids = []
+
+        self.check_access_rights("read")
+        self.check_access_rule("read")
+
+        res = []
+
+        # Prefetch the fields used by the `name_get`
+        self.sudo().read(
+            ['name', 'default_code', 'product_tmpl_id',
+             'attribute_value_ids', 'attribute_line_ids'], load=False)
+
+        product_template_ids = self.sudo().mapped('product_tmpl_id').ids
+
+        if partner_ids:
+            supplier_info = self.env['product.supplierinfo'].sudo().search([
```

Comment:
You mustn't search for supplierinfo here. Better to pass a context here and on search of the supplierinfo, mix results of both supplierinfo + customerinfo.

Reply:
odoo doesn't let you mix model records. It will say you "mixing orange with apples" message.

Reply:
I know, but you only need to return an iterable, so deal with that doing a list of recordsets of one element.

Reply:
no, in the odoo product `name_get` code, after doing the `search` in the supplierinfo model, it does a `read` on the result. If the result is a list as you propose, then it will break when trying to do the `read`. see: https://github.com/OCA/OCB/blob/12.0/addons/product/models/product.py#L458

Reply:
OK, and what about returning new supplierinfo recordsets, but with customerinfo fields? Does the read works on them?

Reply:
You might trick the `read` also if not.